modules/markup/markdown/markdown_test.go

```diff
@@ -509,15 +509,23 @@ func TestMathBlock(t *testing.T) {
 		},
 		{
 			`$a a$b b$`,
-			`<p><code class="language-math is-loading">a a$b b</code></p>` + nl,
+			`<p><code class="language-math is-loading">a a</code>b b$</p>` + nl,
 		},
 		{
 			`a a$b b`,
 			`<p>a a$b b</p>` + nl,
 		},
 		{
 			`a$b $a a$b b$`,
-			`<p>a$b <code class="language-math is-loading">a a$b b</code></p>` + nl,
+			`<p>a<code class="language-math is-loading">b </code>a a<code class="language-math is-loading">b b</code></p>` + nl,
+		},
+		{
+			"a$x$",
+			`<p>a<code class="language-math is-loading">x</code></p>` + nl,
+		},
+		{
+			"$x$a",
+			`<p><code class="language-math is-loading">x</code>a</p>` + nl,
```

Comment:
Sorry, the new tests do not seem right.

`$x$a` shouldn't be considered as math expr IMO

Reply:
It seems that GitHub doesn't allow preceding/succeeding alphanumeric characters, GitLab's behaviour seems a bit "irregular" since it allows succeeding alphabetical characters but not numerical characters.

May i ask why is it that `$x$a` shouldn't be considered a math expression, and what would be the correct behaviour for the rendering in that case?

Reply:
That's just my personal feeling, if it is possible I think it's better to use some stricter syntax. If `$x$a` is considered as a math expr, then something like `The price is between US$1 and US$10` will be rendered incorrectly? (haven't really tested, just a guess)

Reply:
I see now, the behaviour you described will indeed render incorrectly. 
![image](https://github.com/go-gitea/gitea/assets/114936010/2e0e64b1-3802-47ae-a953-899ed333668d)
I will change my code to accept only preceding characters but not succeeding characters.

Reply:
[fff63a7](https://github.com/go-gitea/gitea/pull/30175/commits/fff63a71a525256839b42e23bc43a6475ade0067)

It's implemented here.